### DISCUSSION TITLE
2 search results webparts, 2 similar queries, exclude 1 result if already present in other webpart

### DISCUSSION THREAD

**Discussion by paxvas** [microsoft-search/pnp-modern-search#2442] [Q&A]
Hi,

I have a requirement to have 2 search result webparts on 1 page with overlapping results. Would it be possible to exclude results from one webpart, when it is already present in the second webpart? I can't achieve what eaxctly  I want via the query itself as the customer wants something special. Any suggestions how to do that? Something like connected webparts with filtering. THX for any hints.

**Comment by wobba** [comment]
Not the way you want I believe. Explain your scenario and someone might be able to come up with a solution.

**Reply by paxvas** [reply]
hi Mikael,

the requirement is based on an intranet homepage with different webparts szenario. 

On top of the page there should be a pnp search webpart which aggregates 5 news of type Topstory (example query - contenttype:news topstory:1).

Below we have a webpart showing the latest news (example query - contenttype:news)

The requirement is that if the customer adds an 6 Topstory the oldest disappears from the webpart on top but should be visible in the webpart below. The customer does NOT want to see duplicated news in the latest news webparts. 

In the latest news webpart I just query for news so i will see this duplicates; So I need to get rid of them in a why. Basically the latest news webpart should show news minus visible top stories from the webpart above. 

For further question I am availabe. Thank you again for your year long help through all this topics.

Kind regards Janko

**Reply by wobba** [reply]
If you use `contenttype:news NOT topstory:1` in the second web part, that should exclude what you have in the top one. Will that work?

**Comment by paxvas** [comment]
Hi,

That was my idea.It will not work as they want to see also topstories in
the second webpart, but only these which are Not present already in webpart
1. It's like result 2 minus result 1= correct.

Mikael Svenson ***@***.***> schrieb am Mo., 24. Okt. 2022,
08:18:

**Reply by wobba** [reply]
Not possible. You should instead use one web part and do something link this in the query template to get top stores on top, and the rest below.

Query template: `SPContentType:news`

Sorting: TopStory:descending +  LastmodifiedTime:descending

Depending on what type of column TopStory is. Or you can play with XRANK (but use the nb) parameter similar to this post I wrote some time ago https://www.techmikael.com/2016/12/tip-for-writing-multiple-xranks-or.html
